apps/itest/test/itest/batch_transactions_2_test.exs

```diff
@@ -0,0 +1,376 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule BatchTransactions2Tests do
+  use Cabbage.Feature, async: true, file: "batch_transactions_2.feature"
+  @moduletag :transactions
+
+  require Logger
+
+  alias ChildChainAPI.Api.Transaction
+  alias ChildChainAPI.Connection, as: ChildChain
+  alias ChildChainAPI.Model.TransactionBatchSubmitBodySchema
+  alias ExPlasma.Transaction.Payment
+  alias Itest.Account
+  alias Itest.ApiModel.WatcherSecurityCriticalConfiguration
+  alias Itest.Client
+  alias Itest.Fee
+  alias Itest.Poller
+  alias Itest.Transactions.Currency
+  alias Itest.Transactions.Encoding
+
+  setup do
+    {:ok, _} = DebugEvents.start_link()
+
+    [{alice_address, alice_pkey}, {bob_address, bob_pkey}, {eve_address, eve_pkey}] = Account.take_accounts(3)
+
+    eth_fee =
+      Currency.ether()
+      |> Encoding.to_hex()
+      |> Fee.get_for_currency()
+      |> Map.get("amount")
+
+    %{
+      "fee" => eth_fee,
+      "Alice" => %{
+        address: alice_address,
+        pkey: "0x" <> alice_pkey,
+        gas: 0,
+        ethereum_balance: 0,
+        ethereum_initial_balance: 0,
+        child_chain_balance: 0,
+        utxos: [],
+        transaction_submit: nil,
+        receipt_hashes: []
+      },
+      "Bob" => %{
+        address: bob_address,
+        pkey: "0x" <> bob_pkey,
+        gas: 0,
+        ethereum_balance: 0,
+        ethereum_initial_balance: 0,
+        child_chain_balance: 0,
+        utxos: [],
+        transaction_submit: nil,
+        receipt_hashes: []
+      },
+      "Eve" => %{
+        address: eve_address,
+        pkey: "0x" <> eve_pkey,
+        gas: 0,
+        ethereum_balance: 0,
+        ethereum_initial_balance: 0,
+        child_chain_balance: 0,
+        utxos: [],
+        transaction_submit: nil,
+        receipt_hashes: []
+      }
+    }
+  end
+
+  defwhen ~r/^they deposit "(?<amount>[^"]+)" ETH to the root chain$/,
+          %{amount: amount},
+          state do
+    entity_1 = "Alice"
+    entity_2 = "Bob"
+    entity_3 = "Eve"
+
+    receipts =
+      Enum.map([entity_1, entity_2, entity_3], fn entity ->
+        %{address: address} = state[entity]
+
+        {:ok, receipt_hash} =
+          amount
+          |> Currency.to_wei()
+          |> Client.deposit(address, Itest.PlasmaFramework.vault(Currency.ether()))
+
+        receipt_hash
+      end)
+
+    # lets wait for the three deposits to be recognized
+    geth_block_every = 1
+
+    {:ok, response} =
+      WatcherSecurityCriticalAPI.Api.Configuration.configuration_get(WatcherSecurityCriticalAPI.Connection.new())
+
+    watcher_security_critical_config =
+      WatcherSecurityCriticalConfiguration.to_struct(Jason.decode!(response.body)["data"])
+
+    finality_margin_blocks = watcher_security_critical_config.deposit_finality_margin
+    to_miliseconds = 1000
```

Comment:
I think you can get rid of the `geth_block_every` var and just explain everything in the comment for this variable